firebase_admin/project_management.py

```diff
@@ -0,0 +1,495 @@
+            app_class=IosApp)
+
+    def _create_app(
+            self,
+            platform_resource_name,
+            identifier_name,
+            identifier_label,
+            identifier,
+            display_name,
+            app_class):
+        """Creates an Android or iOS App."""
+        _check_is_string_or_none(display_name, 'display_name')
+        path = '/v1beta1/projects/{0}/{1}'.format(self._project_id, platform_resource_name)
+        request_body = {'displayName': display_name, identifier_name: identifier}
+        response = self._make_request('post', path, identifier, identifier_label, json=request_body)
+        operation_name = response['name']
+        try:
+            poll_response = self._poll_app_creation(operation_name)
+            return app_class(app_id=poll_response['appId'], service=self)
+        except _PollingError as error:
+            raise ApiCallError(
+                self._extract_message(operation_name, 'Operation name', error), error)
+
+    def _poll_app_creation(self, operation_name):
+        """Polls the Long-Running Operation repeatedly until it is done with exponential backoff."""
+        for current_attempt in range(_ProjectManagementService.MAXIMUM_POLLING_ATTEMPTS):
+            delay_factor = pow(
+                _ProjectManagementService.POLL_EXPONENTIAL_BACKOFF_FACTOR, current_attempt)
+            wait_time_seconds = delay_factor * _ProjectManagementService.POLL_BASE_WAIT_TIME_SECONDS
+            time.sleep(wait_time_seconds)
+            path = '/v1/{0}'.format(operation_name)
+            poll_response = self._make_request('get', path, operation_name, 'Operation name')
+            done = poll_response.get('done')
+            if done:
+                response = poll_response.get('response')
+                if response:
+                    return response
+                else:
+                    raise _PollingError('Operation terminated in an error.')
+        raise _PollingError('Polling deadline exceeded.')
+
+    def _make_request(self, method, url, resource_identifier, resource_identifier_label, json=None):
+        try:
+            return self._client.body(method=method, url=url, json=json, timeout=self._timeout)
+        except requests.exceptions.RequestException as error:
+            raise ApiCallError(
+                self._extract_message(resource_identifier, resource_identifier_label, error), error)
+
+    def _extract_message(self, identifier, identifier_label, error):
+        if not isinstance(error, requests.exceptions.RequestException) or error.response is None:
+            return str(error)
+        status = error.response.status_code
+        message = _ProjectManagementService.ERROR_CODES.get(status)
+        if message:
+            return '{0} "{1}": {2}'.format(identifier_label, identifier, message)
+        return '{0} "{1}": Error {2}.'.format(identifier_label, identifier, status)
+
+    def _none_to_empty(self, string):
```

Comment:
Make this a static or class method. Basically any helper methods that don't need `self` can be static or class method.

Reply:
Done.